examples/models/Question_Answering/PyPandaQA.py

```diff
@@ -0,0 +1,560 @@
+    scores = [getSingleLOEScore(paper_idx, idx) for idx in range(len(idx_design_map))]
+    sorted_indexes = np.argsort(scores)[::-1]
+    for sorted_index in sorted_indexes:
+        if scores[sorted_index] >= minimum_occurrencies[sorted_index]:
+            return idx_design_map[sorted_index]
+    return "Information not available"
+
+'''def getFullDictQueries(data, num_papers=5): # predict
+    final_dict = {}
+    for task in data.keys():
+        print(f"Processing: {task}")
+        task_questions = data[task]['questions']
+        dict_task_quest ={}
+        dict_queries = {}
+        for idx, query in enumerate(task_questions):
+            print(f"Getting answers from query: {query}")
+            indexes_papers = extractPapersIndexes(query, num_papers=num_papers)
+            dict_queries[query] = getAllContexts(query, indexes_papers)
+        dict_task_quest['queries'] = dict_queries
+        dict_task_quest['task_name'] = data[task]['area']
+        final_dict[task] = dict_task_quest
+    return final_dict'''
+
+
+def answerQuestion(question, paper):
+    """
+    This funtion provides the best answer found by the Q&A model, the chunk containing it
+    among all chunks of the input paper and the score obtained by the answer
+    """
+    inputs = [qa_tokenizer.encode_plus(
+        question, paragraph, add_special_tokens=True, return_tensors="pt") for paragraph in paper]
+    answers = []
+    confidence_scores = []
+    for n, Input in enumerate(inputs):
+        input_ids = Input['input_ids'].to(torch_device)
+        token_type_ids = Input['token_type_ids'].to(torch_device)
+        if len(input_ids[0]) > 510:
+            input_ids = input_ids[:, :510]
+            token_type_ids = token_type_ids[:, :510]
+        text_tokens = qa_tokenizer.convert_ids_to_tokens(input_ids[0])
+        start_scores, end_scores = qa_model(input_ids, token_type_ids=token_type_ids)
+        answer_start = torch.argmax(start_scores)
+        answer_end = torch.argmax(end_scores)
+        # if the start token of the answer is contained in the question, the start token is moved to
+        # the first one of the chunk
+        check = text_tokens.index("[SEP]")
+        if int(answer_start) <= check:
+            answer_start = check + 1
+        answer = qa_tokenizer.convert_tokens_to_string(text_tokens[answer_start:(answer_end + 1)])
+        answer = answer.replace('[SEP]', '')
+        confidence = start_scores[0][answer_start] + end_scores[0][answer_end]
+        if answer.startswith('. ') or answer.startswith(', '):
+            answer = answer[2:]
+        answers.append(answer)
+        confidence_scores.append(float(confidence))
+
+    maxIdx = np.argmax(confidence_scores)
+    confidence = confidence_scores[maxIdx]
+    best_answer = answers[maxIdx]
+    best_paragraph = paper[maxIdx]
+
+    return best_answer, confidence, best_paragraph
+
+class PyPandaQA(PandaTorchBasicModel):
+    """
+    Implementation of PyTorch Vgg
```

Comment:
it is not vgg, may change the name

Reply:
you can see the renamed class https://github.com/nusdbsystem/singa-easy/blob/master/singa_easy/models/TorchModel.py#L74